benchmark/C/Savina/src/Matrix.h

```diff
@@ -0,0 +1,49 @@
+/**
+ * Implementation of the Matrix abstract data type for use in some benchmarks.
+ */
+
+#ifndef MATRIX_H
+#define MATRIX_H
+
+#include <stddef.h>
+#include <stdlib.h>
+
+typedef struct Matrix {
+    int n_cols;
+    double* content;
+} Matrix;
+
+/*
+ * Allocate a new Matrix.
+ */
+Matrix mat_new(size_t m, size_t n) {
+    Matrix ret = {.n_cols = n, .content = (double*) malloc(m * n * sizeof(double))};
+    return ret;
+}
+
+/*
+ * Deallocate the given Matrix.
+ */
+void mat_destroy(Matrix matrix) {
+    free(matrix.content);
+}
+
+/*
+ * Return a pointer to entry (i, j) of the given Matrix.
+ */
+double* mat_at(Matrix matrix, size_t i, size_t j) {
+    return (double*) matrix.content + i * matrix.n_cols + j;
```

Comment:
I think `matrix.content[i * matrix.n_cols + j]` is slightly more readable. Moreover, I would strongly suggest that some form of memory bound check be performed here (e.g., `(i * matrix.size_y + j) < (matrix.size_y * matrix.size_x)`).